### PR TITLE
Multithreading 30/N: HTML5 Event Backproxying

### PR DESCRIPTION
**Pull request by juj**
This is the followup to #6201: refactoring HTML5 API to add an extra parameter to all event callback registration functions which enable caller to specify the context of the thread that the event callback is called in. This enables applications to choose to run event handlers on the main browser thread (synchronizing manually), or to run event handlers in the calling thread, or even in some other pthread.

### CODE REVIEW

File: src/library_html5.js [hunk 1]
```diff
@@ -1911,14 +2191,23 @@ var LibraryJSEvents = {
 
   emscripten_set_webglcontextlost_callback: function(target, userData, useCapture, callbackfunc) {
     JSEvents.registerWebGlEventCallback(target, userData, useCapture, callbackfunc, {{{ cDefine('EMSCRIPTEN_EVENT_WEBGLCONTEXTLOST') }}}, "webglcontextlost");
```
Comment (nickshinpho) [inline src/library_html5.js]: note to self: seems emscripten_set_webglcontextlost_callback: should be deleted?

**Comment by juj** [comment]
Rebased to latest and added `@requires_threads` directives to new tests.

**Comment by dschuff** [comment]
I tested this locally and the tests don't seem to be working for me. When I test with Chrome dev and FF nightly, the  `interactive.test_html5_callbacks_on_calling_thread` test runs and prints the request to move the mouse cursor but doesn't seem to be getting any events. When I run  `interactive.test_html5_callback_on_two_threads` test 1 seems to get the keypress event, prints "test passed!" but then reports a result of 1 instead of 0. Test 2 also seems to get keypress events, but it never terminates, it just keeps receiving and printing keypress output. Do those work for you? Am I just holding it wrong?

**Comment by juj** [comment]
Oh, the issue was that the line `EM_ASM(Module['noExitRuntime'] = true);` in the test was not being honored, and the pthread exited nevertheless. Then the events coming from the main browser thread were getting ignored. I had that fix in another upcoming commit, did not realize the dependency with this test. Updated this PR now to contain that fix as well, and verified it to run on my system.

**Comment by juj** [comment]
Yay, now the tests pass.

**Pull request merged**